karabo_bridge/qt.py

```diff
@@ -0,0 +1,153 @@
+import queue
+from itertools import count
+from secrets import token_hex
+
+import zmq
+from qtpy.QtCore import QObject, QThread, QTimer, Signal, Slot
+
+from .serializer import deserialize
+
+class Worker(QThread):
+    data_queued = Signal()
+
+    def __init__(self, endpoint, sock_type, ctrl_endpoint, queue, stop_after=0,
+                 parent=None):
+        super().__init__(parent)
+        self.endpoint = endpoint
+        self.sock_type = sock_type
+        self.ctrl_endpoint = ctrl_endpoint
+        self.stop_after = stop_after
+        self.queue = queue
+
+    def run(self):
+        ctx = zmq.Context.instance()
+        data_sock = ctx.socket(self.sock_type)
+        data_sock.connect(self.endpoint)
+        if self.sock_type == zmq.SUB:
+            data_sock.setsockopt(zmq.SUBSCRIBE, b'')
+        ctrl_sock = ctx.socket(zmq.PULL)
+        ctrl_sock.bind(self.ctrl_endpoint)
+
+        poller = zmq.Poller()
+        poller.register(data_sock, zmq.POLLIN)
+        poller.register(ctrl_sock, zmq.POLLIN)
+
+        if self.sock_type == zmq.REQ:
+            data_sock.send(b'next')
+
+        for i in count(start=1):
+            ready = [sock for (sock, _) in poller.poll()]
+            if ctrl_sock in ready:
+                _ = ctrl_sock.recv()
+                break
+
+
+            if data_sock in ready:
+                raw_msgs = data_sock.recv_multipart(copy=False)
+                data, metadata = deserialize(raw_msgs)
+                self.queue.put((data, metadata))
+                self.data_queued.emit()
+                if (self.stop_after > 0) and (i >= self.stop_after):
+                    break
+
+            if self.sock_type == zmq.REQ:
+                data_sock.send(b'next')
+
+        # Stop receiving
+        ctrl_sock.close()
+        data_sock.close()
+
+
+class QBridgeClient(QObject):
```

Comment:
Could you add a short docstring to this class? Since it's directed to unexperienced developer, it may be worth giving a few guidelines (what's the endpoint to connect to, ...).